### PR TITLE
Applications in the profile

### PR DESCRIPTION
**Pull request by JulesHerrscher**
# Context
As a user, I could join an association and ask for a staff but there where no possibilities to see the statuses of the applications to events or to associations. We then decided to implement a simple way to track applications in the profile of the user.

# Modification
* Added a screen `Applications.kt` and a composable `ApplicationsItem.kt`to display the applications properly.
* Added a viewModel `ApplicationsViewModel.kt` for fetching the required informations from the database.
* Made changed to the database implementation `DbServiceImpl.kt`.
* changed 'NavigationGraph.kt`accordingly as well as the `Profile.kt` page to include my new page.
* added the first tests

# Testing
Did manual testing on the associations only for the moment because of a bug on the main branch + added first connected tests that shows the implementation works.

# Result
profile_page
Applications

### CODE REVIEW

File: app/src/androidTest/java/com/swent/assos/ApplicationsTest.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        .document(assoId)
+        .set(
+            mapOf(
+                "acronym" to "Polylan",
+                "fullname" to "Polylan",
+                "url" to "https://polylan.ch",
+                "description" to "Polylan is the biggest lan party in Switzerland",
+                "logo" to "https://polylan.ch/logo.png",
+                "banner" to "https://polylan.ch/banner.png"))
+
```
Comment (alicepotter) [inline app/src/androidTest/java/com/swent/assos/ApplicationsTest.kt:57]: If I am not mistaken, all the association are already on the firestore database. Hence, you do not need to set the content of the association.

**Review by alicepotter — CHANGES_REQUESTED** [review]
Great PR! I was just wondering what happens when a member deletes an application. Perhaps we should notify the user that their request was deleted and therefore refused. Currently, a member can accept or reject a request, but they can also delete it. Maybe we should only allow a member to delete a request after making a decision about their application. We should discuss this at the next group meeting.

File: app/src/androidTest/java/com/swent/assos/ApplicationsTest.kt (hunk 1 above)
Comment (JulesHerrscher) [reply app/src/androidTest/java/com/swent/assos/ApplicationsTest.kt:57]: Thank you, I fixed it.

**Review by alicepotter — APPROVED** [review]
Perfect!

**Pull request merged**